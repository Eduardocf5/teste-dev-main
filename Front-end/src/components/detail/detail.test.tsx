import { BrowserRouter } from "react-router-dom";
import renderer from 'react-test-renderer';
import Detail from '.';

const value = {
    id: 1,
    name: "Pica pau",
    image: "https://img.elo7.com.br/product/zoom/3FFF8A3/matriz-de-bordado-pica-pau-matriz-picapau.jpg",
    status: "Alive",
    species: "Picidae",
    origin_name: "Campephilus melanoleucos",
    gender: "Male",
    location_name: "Mundo todo",
    type: "Ave",
}

test('render test', () => {
    const component = renderer.create(
        <BrowserRouter>
            <Detail data={value} />
        </BrowserRouter>
    ).toJSON();

    expect(component).toMatchSnapshot();
  });