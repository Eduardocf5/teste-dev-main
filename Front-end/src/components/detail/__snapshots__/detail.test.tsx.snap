// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render test 1`] = `
<div
  className="flex justify-center"
>
  <div
    className="bg-white p-4 rounded-md"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        alt="back-arrow"
        className="w-10 cursor-pointer"
        src="arrow-back.svg"
      />
    </a>
    <div
      className="flex items-center flex-wrap gap-9 mt-5"
    >
      <img
        alt="icond"
        className="rounded-full max-w-44"
        src="https://img.elo7.com.br/product/zoom/3FFF8A3/matriz-de-bordado-pica-pau-matriz-picapau.jpg"
      />
      <div>
        <div
          className="text-4xl"
        >
          Pica pau
        </div>
        <div
          className="flex items-center gap-1"
        >
          <div
            className="rounded-full bg-green-600 w-3 h-3"
          />
          <div>
            Alive
             - 
            Picidae
          </div>
        </div>
      </div>
    </div>
    <div
      className="mt-4"
    >
      <span
        className="font-semibold"
      >
        Nome de origem: 
      </span>
      <span>
        Campephilus melanoleucos
      </span>
    </div>
    <div
      className="mt-2"
    >
      <span
        className="font-semibold"
      >
        Genero: 
      </span>
      <span>
        Male
      </span>
    </div>
    <div
      className="mt-2"
    >
      <span
        className="font-semibold"
      >
        Localização: 
      </span>
      <span>
        Mundo todo
      </span>
    </div>
    <div
      className="mt-2"
    >
      <span
        className="font-semibold"
      >
        Type: 
      </span>
      <span>
        Ave
      </span>
    </div>
  </div>
</div>
`;
